modules/core/src/main/java/org/apache/synapse/mediators/eip/splitter/IterateMediator.java

```diff
@@ -142,11 +146,29 @@ public boolean mediate(MessageContext synCtx) {
                              " messages for processing in sequentially"));
                 }
 
-                MessageContext itereatedMsgCtx =
+                MessageContext iteratedMsgCtx =
                         getIteratedMessage(synCtx, msgNumber++, msgCount, envelope, (OMNode) o);
                 ContinuationStackManager.
-                        addReliantContinuationState(itereatedMsgCtx, 0, getMediatorPosition());
-                target.mediate(itereatedMsgCtx);
+                        addReliantContinuationState(iteratedMsgCtx, 0, getMediatorPosition());
+
+                if(target.isAsynchronous()) {
+                    target.mediate(iteratedMsgCtx);
```

Comment:
Don't we need to add try catch block to this section of the code. If something goes wrong here (SynapseException) it will not be captured.